receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -143,16 +146,36 @@ func (t *transaction) Append(_ storage.SeriesRef, ls labels.Labels, atMs int64,
 		return 0, nil
 	}
 
-	curMF := t.getOrCreateMetricFamily(getScopeID(ls), metricName)
-	err := curMF.addSeries(t.getSeriesRef(ls, curMF.mtype), metricName, ls, atMs, val)
+	curMF, existing := t.getOrCreateMetricFamily(getScopeID(ls), metricName)
+
+	if t.enableNativeHistograms && curMF.mtype == pmetric.MetricTypeExponentialHistogram {
+		curMF.mtype = pmetric.MetricTypeHistogram
+	}
+
+	seriesRef := t.getSeriesRef(ls, curMF.mtype)
+	err := curMF.addSeries(seriesRef, metricName, ls, atMs, val)
 	if err != nil {
-		t.logger.Warn("failed to add datapoint", zap.Error(err), zap.String("metric_name", metricName), zap.Any("labels", ls))
+		// Handle special case of float sample indicating staleness of native
```

Comment:
trying to wrap my head around this.  Why does the empty label error indicate is is a stale exponential histogram?

Reply:
I've added more comment, ptal

Reply:
makes sense, thanks.